### PR TITLE
Fixes #1395

### LINKED ISSUE #1395: Improve Arduino Generator
Attached you will find a generated and adapted code from this simple Infomodel https://vorto.eclipse.org/#/details/com.bosch.iotacademy.demo:Illuminance:1.0.0 

In it, I have made the following generator relevant changes: all line numbers reference the attached version of the code ;)

1 rearranged configuration order (lines 14 - 46)
2 moved hono_authId definition to configuration (from line 194 to 36)
3 added comment on var name of sample numeric value (lines 48-54 and roughly 237)
4 uppercased fingerprint placeholder (line 25)
5 added more verbose serial output to setup_wifi() and corrected typo in comment (lines 103-115)
6 cleaned up first comment in setup()-function, necessary due to 2 (line 192) 
7 changed telemetry topic to append deviceId instead of clientId, to be more in line with Hono/Hub documentation (line 198-199)
8 corrected typo in publish data header comment (line 96)

Additional Info in Tutorial needed for MAC Users:
The ESP32 Development board's ports will not be recognized by the Arduino IDE 
since there are several drivers that are missing on a Mac. 
1.	Download the drivers through the SiLabs website: https://www.silabs.com/Support%20Documents/Softwar...
2.	Install the drivers onto the computer
3.	Plug in the ESP32 through the micro-usb cable to the computer, and the port will appear on the Arduino IDE under Tools>>Ports.

### PR DESCRIPTION
**Pull request by aedelmann**

### CODE REVIEW

**Comment by aedelmann** [comment]
@jacewski Can you please review the code changes.

File: generators/generator-eclipsehono/hono-arduino/src/main/java/org/eclipse/vorto/codegen/hono/arduino/ArduinoSketchTemplate.xtend [hunk 1]
```diff
@@ -208,13 +214,12 @@ class ArduinoSketchTemplate extends ArduinoTemplate<InformationModel> {
 		  
 		    /* Create a the MQTT client and the Hono authID*/
```
Comment (jacewski) [inline generators/generator-eclipsehono/hono-arduino/src/main/java/org/eclipse/vorto/codegen/hono/arduino/ArduinoSketchTemplate.xtend]:
```suggestion
		   /* Create a MQTT client ID */
```

File: generators/generator-eclipsehono/hono-arduino/src/main/java/org/eclipse/vorto/codegen/hono/arduino/ArduinoSketchTemplate.xtend [hunk 2]
```diff
@@ -208,13 +214,12 @@ class ArduinoSketchTemplate extends ArduinoTemplate<InformationModel> {
...
 		    hono_clientId = hono_deviceId;
-		    hono_authId = String(hono_clientId) + "@" + String(hono_tenant);
 		    
 		    Serial.print("Device ID: ");
 		    Serial.println(hono_clientId);
 		    
 		    /* Add the client ID to the telemetry topic as the final element */
```
Comment (jacewski) [inline generators/generator-eclipsehono/hono-arduino/src/main/java/org/eclipse/vorto/codegen/hono/arduino/ArduinoSketchTemplate.xtend]:
```suggestion
		    /* Add the device ID to the telemetry topic as the final element */
```

**Comment by aedelmann** [comment]
Hi @jacewski 
I do not quite understand the meaning behind the sample_value / dummy_value. Why is this generated at all ? Wouldn't it just need to generate the fields coming from the information model ?

**Comment by jacewski** [comment]
Hi @aedelmann,
the dummy value is used as fake data for all generated number and text fields. This makes it possible to test the connection and payload configuration without wiring up any of the actual sensors.
Which for prototyping is a kind of a good thing :)

**Pull request closed without merging**